### ISSUE TITLE
Forward email as attachment

### ISSUE THREAD

**Issue by AuspeXeu** [nodemailer/wildduck-webmail#16]
Is there a plan to implement this feature? :)

**Comment by bunyevacz** [comment]
Is it a duplicate of issue #6 ?

So the problem for you to not be able to forward email *with* attachment, 
or forward email as a file attached to the new email á la outlook's .eml file?

**Comment by AuspeXeu** [comment]
The latter, so no duplicate :)

**Comment by bunyevacz** [comment]
You need to elaborate a bit more what is the usecase here in general, and what is your specific usecase too.
It's not up to me to decide on future development directions or even a future feature, I just want to have a clear picture why and how is it needed.

First, gmail does not support .eml file creation. When you forward an email in gmail, you get the original email content quoted ( `>` char) and the original email's attachment reattached to the email. 
It is not working either atm, see bug #6.

Let's assume, you get an email, which you want to forward as an .eml file (just what you requested).
The email was written in outlook, so it is an html email. It also contains a company signature, so 
there is a image001.png and image003.jpg attached, and shown as a stylized signature at the end of email.
The email itself has a preview of the new company flyer embedded into the html. So it is also attached as .jpg file, lets say meandme_flyer.jpg.
And the flyer itself is attached to the email as .pdf, and as .docx.

Now you forward it the whole thing as a .eml file. The other end using gmail can not even open the whole thing. You need an outlook to see what was the content. 
What is the point? I really do not get it.

**Comment by AuspeXeu** [comment]
Thanks for taking the time for this detailed comment @bunyevacz much appreciated :)

Basically we have the following use case in our company.
Person `A` receives an email `e`, now `A` realises person `B` should answer `e`.
In order to make it the easiest for person `B`, the original `e` gets forwarded as attachment so that `B` can reply directly. In this scenario `A` can also add some info `B` might need to process `e` by writing it into the forwarding email's body.

Does this make sense? Thanks!

**Comment by bunyevacz** [comment]
We do the following:

1. Person `A` receives `e` from `O`. It realizes, that person `B` can make a competent decision (eg. payment request).
2. Person `A` forwards `e` to person `B`, also **CC**ing the original `O`.
3. Person `B` receives `e'`, and it replies to it.
4. Both person `A` and person `O` receives the reply.

Otherwise person `O` would be confused *why* person `B` responds instead of person `A`!

If the `B -> O` is confidential and `A` should not be informed, then `B`, simply removes `A` from **CC** field.
If `O` should believe that `B` is responding to him directly *without* `A` receiving the reply, then `B` moves the **CC** field to **BCC** field.

I think it is a better method, in your case `O` always wonders why `B` responds instead of `A`.

If person `A` quits the company, then you can make forwarding emails to person `B`. It is working in wildduck at the moment already. That case may need to be finetuned regarding GDPR in Europe. Ie. it should include an autoreply to let know `O`, that person `A` no longer works at the company.
I have seen lately, that if person `A` quits the company, they make a simple autoreply in the account of `A`, 
with the following content: ` 'A' no longer works with us, please resend your message to b@weareawesome.com. Thank you`.
I always wondered, how many people sends their message one more time. Or just ignores the auto-reply.

Still .eml handling is problematic, and it always was.

**Comment by AuspeXeu** [comment]
I understand the idea of your workflow. But let me explain what the rational behind ours is :)

Outwards facing we have one unified email address let's say `weareawesome@company.com`. Whenever an email leaves the company the sender's reply to gets set to that address. This way, no one cares if someone quits or whatever. (Note: Yes, this only works bc we are a small company with not too many incoming mails so that we can handle a single reply to) Regarding your point that `0` cannot open it that is not the case. We forward the original email as attachment, employee `B` opens the attachment and processes it _as if_ he received it himself/directly in the first place. That way, there is also no need to delete additional comments from `A`, which `B` might forget to delete which then looks bad if `0` receives internal comments. Also, `B` does not have to worry about any **CC** or **BCC**.

**Comment by bunyevacz** [comment]
Ok, then could you please post a simple email (you can create one with whatever content you like) preferrably zipped/tar.gz up, and put out to the internet to see?
Please include at least one attachment.

I'm asking this, because there are already some infrastructure in place in wildduck/wildduck-webmail.
So when you open an email in wildduck-webmail, there is a `Details->Original message` button in the top right corner (just above the email), which sends a request like:
```
view email normally: /webmail/2b184657c4acb0049beaedc6/message/5818
view email as .eml file: /webmail/2b184657c4acb0049beaedc6/raw/5818.eml
```
So the backend (wildduck) is up to the job (create .eml file), the frontend (wildduck-webmail) needs to adjust slightly, and also implement issue #6. 
Then rather then the original attachment gets attached, the generated .eml file gets attached.

Seems like we could implement it together with issue #6. If there is a real usecase here... :-)
Still seems an outlook only workflow. What are you using as clients?

**Comment by AuspeXeu** [comment]
Sure, I can provide an example. I do agree it is an outlook only workflow in the sense that it can only be done using outlook at the moment. Nevertheless I do think it's actually quite nice for the scenario I described. There is not much work to do in terms of changing addresses etc. and it's quite resilient against employees messing up/forgetting things. It's very straight forward for them as to how to handle an email that has been delegated to them.

**Comment by bunyevacz** [comment]
Yeah, except many viruses [spread][0] as `.eml` files :(

[0]: https://www.google.com/search?client=ubuntu&channel=fs&q=.eml+viruses&ie=utf-8&oe=utf-8

**Comment by AuspeXeu** [comment]
I guess there is always a downside...

**Comment by bunyevacz** [comment]
yes, if you stay on wildduck-webmail only, then processing .eml file does not care much threat.
But right now wildduck-webmail lacks some key features (draft autosave, attachment forwarding, etc) to be used as the only email client imho. 

In either case, if @andris9 agrees with me, I think we can close this issue, and continue in issue #6.
So issue #6 is extended not only forward with attachments, but forward emails as .eml file. 

Still an example forwarded mail (what you use in your company) would be much helpful.

**Comment by AuspeXeu** [comment]
Oh, before I forget. Obviously wildduck-webmail would have to support opening those `.eml` attachments within the browser and and offer for example a reply operation. Not sure that came across from my previous comments.

**Comment by AuspeXeu** [comment]
@bunyevacz what format would you like me to provide the example?

**Comment by bunyevacz** [comment]
how you usually send it inside your company zipped up.
an html email with an attachment.

**Comment by AuspeXeu** [comment]
I saved it as `.eml` using thunderbird. Hopefully this helps.

https://gist.github.com/AuspeXeu/a37d4a2bfc901dfb6f1d2b6ccb18d828
